src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -16,14 +16,13 @@
 /** Class controlling the TaskData object. */
 public final class TaskController {
 
-  private HashSet<TaskData> tasks;
+  private DatastoreService datastore;
```

Comment:
Good move removing the `tasks` member since the controller doesn't currently need to keep track of state anyways. (It can be added later on if there's a good use case for holding on to it).

I would suggest moving the Datastore ownership out of this class though, or at least initialize it with a copy from outside.  For some database engines, each instance of these would normally hold on to a connection to the database.  Meaning multiple instances would create multiple open connections.  Normally it would be best practice for applications to hold on to one open connection and share it throughout the application (by passing it around as params).  This saves resources by minimizing unneeded open connections.  It also ensures all usage are connected to the same exact database (in cases where you have multiple databases available to connect to).